src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,285 @@
+"""
+Implementation of scoring functions that are consistent for
+single-valued forecasts targeting quantiles, expectiles or Huber functionals.
+"""
+
+from typing import Callable, Optional
+
+import xarray as xr
+
+from scores.functions import apply_weights
+from scores.typing import FlexibleDimensionTypes
+from scores.utils import gather_dimensions
+
+
+def consistent_expectile_score(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    alpha: float,
+    phi: Callable[[xr.DataArray], xr.DataArray],
+    phi_prime: Callable[[xr.DataArray], xr.DataArray],
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Calculates the score using a scoring function that is consistent for the
+    alpha-expectile functional, based on a supplied convex function phi.
+    See Geniting (2011), or Equation (10) from Taggart (2022).
+
+    .. math::
+
+        S(x, y) =
+        \\begin{cases}
+        (1 - \\alpha)(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & y < x \\\\
+        \\alpha(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & x \\leq y
+        \\end{cases}
+
+    where
+        - :math:`x` is the forecast
+        - :math:`y` is the observation
+        - :math:`\\alpha` is the expectile level
+        - :math:`\\phi` is a convex function of a single variable
+        - :math:`\\phi'` is the subderivative of :math:`\\phi`
+        - :math:`S(x,y)` is the score.
+
+    Note that if :math:`\\phi` is differentiable then :math:`\\phi'` is its derivative.
+
+    The score is negatively oriented with :math:`0 \\leq S(x,y) < \\infty`.
+
+    We strongly recommend that you work through the tutorial 
+    https://scores.readthedocs.io/en/stable/tutorials/Consistent_Scores.html to help you 
+    understand how to use the consistent scoring functions in scores.
```

Comment:
```suggestion
    understand how to use the consistent scoring functions in `scores`.
```
I am assuming backticks render as expected in the API rendering in readthedocs? If not, ignore this (and the next two suggestions). 

If it will render as expected, I think it might help with clarity in these cases. (Unless I have totally misunderstood, in which case also please ignore).

I also don't know if there is a preferred way of annotating this in the docstrings for people reading the actual code (rather than in readthedocs). So if this isn't appropriate, or there is a better way to do it, then please ignore my suggestions.